sergeitoptalo/nodejs-aws-be
# Task 09

Application:
https://daqzoqrr7tgx6.cloudfront.net

FE changes:
https://github.com/sergeitoptalo/nodejs-aws-fe/pull/7 

- [x] A working and correct express application should be in the bff-service folder.
- [x] - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service. The response from the bff-service should be the same as if the recipient service was called directly.

Products endpoint:
https://o1oewsft1f.execute-api.eu-west-1.amazonaws.com/dev/products

Cart endpoint:
https://1qb63bxdq6.execute-api.eu-west-1.amazonaws.com/dev/api/profile/cart

Products with BFF (http):
http://sergeitoptalo-bff-api-dev.eu-west-1.elasticbeanstalk.com/product-service/products

Cart with BFF (http):
http://sergeitoptalo-bff-api-dev.eu-west-1.elasticbeanstalk.com/cart-service/profile/cart

Products with BFF (https):
https://jl32md4xz7.execute-api.eu-west-1.amazonaws.com/dev/product-service/products

Cart with BFF (https):
https://jl32md4xz7.execute-api.eu-west-1.amazonaws.com/dev/cart-service/profile/cart

- [x] Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
To test cache:
- create new product here: https://daqzoqrr7tgx6.cloudfront.net/admin/product-form
- check list of products https://daqzoqrr7tgx6.cloudfront.net/admin/products (product should not be on the list)
- wait for 2 minutes
- check list of products again - product should be there.
